dev-client/src/store/sync/hooks/syncHooks.tsx

```diff
@@ -71,3 +77,21 @@ export const useRetryInterval = (interval: number, action: () => void) => {
 
   return {beginRetry, endRetry};
 };
+
+export const useSyncErrorSiteIds = () => {
+  return Object.keys(useSelector(selectSyncErrorSites));
+};
+
+export const usePullDispatch = () => {
+  const dispatch = useDispatch();
+
+  const currentUserID = useSelector(selectCurrentUserID);
+
+  return useCallback(() => {
+    if (currentUserID !== undefined) {
```

Comment:
I think it would be better to have the logic around `currentUserId` exist outside of the hook here. Similarly I'm a little uncomfortable with the coupling of dispatching `setPullRequested` and `fetchSoilDataForUser`; I think it'd be more appropriate to have each `dispatch` in its own mini hook and still use the logic in the component to coordinate them + branch on the user ID. This is because these hooks files aren't covered by tests, and we are using them to make the actual logic of these processes easier to test. (Happy to discuss further if this is a confusing comment).

Reply:
That makes sense, thanks!

Reply:
FYI my last 3 commits should address your 3 comments, lmk if I misunderstood anything though!